File: python3/folding.py
# vim: set nospell:
# vim: set foldmethod=indent:
import vim
import re


def OrgFolding():
    lnumcons = int(vim.vvars['lnum'])
    lnum = vim.current.buffer[lnumcons-1]
    nextline = vim.current.buffer[lnumcons]
    global ntcodeblockcheap
    if re.match(r'^\s*:PROPERTIES:', lnum):
        return 'a7'
    elif re.match(r'^\s*:END:', lnum):
        return "s7"
    elif re.match(r'^\s*#\+BEGIN_SRC', lnum):
        ntcodeblockcheap = False
        return "a7"
    elif re.match(r'^\s*#\+BEGIN', lnum):
        return "a7"
    elif re.match(r'^\s*#\+END_SRC', lnum):
        ntcodeblockcheap = True
        return "s7"
    elif re.match(r'^\s*#\+END', lnum):
        return "s7"
    elif re.match(r'#.*\{\{\{', lnum):
        return "a7"
    elif re.match(r'# \}\}\}', lnum):
        return "s7"
    elif re.match(r'^\*+ ', lnum) and ntcodeblockcheap:
        return ">" + str(lnum.find(' ')) 
    elif re.match(r'^.+$', lnum) and re.match(r'^=+$', nextline) and ntcodeblockcheap:
        return ">1"
    elif re.match(r'^.+$', lnum) and re.match(r'^-+$', nextline) and ntcodeblockcheap:
        return ">2"
    else:
        return "="


def NtcodeblockPy(x):
    result = vim.eval(f"synIDattr(synID({x}, 1, 1), 'name') !=# 'OrgCodeBlock'")
    return result=='1'

File: lua/folding.lua
local M = {}

local function M.Ntcodeblock(x)
  return vim.fn.synIDattr(vim.fn.synID(x, 1, 1), "name") ~= "OrgCodeBlock"
end

local function M.OrgFolding()
  local line = vim.fn.getline(vim.v.lnum) -- Obtener la línea actual
  local nextline = vim.fn.getline(vim.v.lnum + 1) -- Obtener la línea siguiente

  if line:match("^%s*:PROPERTIES:$") then
    return "a7"
  end
  if line:match("^%s*:END:$") then
    return "s7"
  end
  if line:match("#%+BEGIN_") then
    return "a7"
  end
  if line:match("#%+END_") then
    return "s7"
  end
  if line:match(" {{{") then
    return "a7"
  end
  if line:match(" }}}") then
    return "s7"
  end
  if line:match("^%*+ ") and Ntcodeblock(vim.v.lnum) then
    return ">" .. (line:find(" ") or 0) -- Encontrar el primer espacio
  end
  if line:match("^.+$") and nextline:match("^=+$") and Ntcodeblock(vim.v.lnum + 1) then
    return ">1"
  end
  if line:match("^.+$") and nextline:match("^%-+$") and Ntcodeblock(vim.v.lnum + 1) then
    return ">2"
  end
  return "="
end

return M
